### PR TITLE
Adjust pod security for V8o requirements

### PR DESCRIPTION
**Pull request by rjeberhard**

### CODE REVIEW

File: documentation/4.0/content/security/domain-security/pod-and-container.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  capabilities:
+    drop:
+    - ALL
+```
+
+On OpenShift environments, the operator omits the `runAsUser` and `runAsGroup` elements.
+
+Customers can [configure pod and container generation](https://oracle.github.io/weblogic-kubernetes-operator/managing-domains/domain-resource/#domain-and-cluster-spec-elements)
+for WebLogic Server instances using the `serverPod` element in the Domain resource. If specified, the operator will use the
+`serverPod.podSecurityContext` or `serverPod.containerSecurityContext` content from the Domain resource rather than using the default content shown above.
```
Comment (rosemarymarano) [inline documentation/4.0/content/security/domain-security/pod-and-container.md:38]: shown above. -> shown previously.

File: documentation/domains/Cluster.md [hunk 2]
```diff
@@ -56,7 +56,7 @@ The specification of the operation of the WebLogic cluster. Required.
 | `affinity` | [Affinity](k8s1.13.5.md#affinity) | The Pod's scheduling constraints. More info: https://oracle.github.io/weblogic-kubernetes-operator/faq/node-heating/.  See `kubectl explain pods.spec.affinity`. |
 | `annotations` | Map | The annotations to be added to generated resources. |
 | `containers` | Array of [Container](k8s1.13.5.md#container) | Additional containers to be included in the server Pod. See `kubectl explain pods.spec.containers`. |
-| `containerSecurityContext` | [Security Context](k8s1.13.5.md#security-context) | Container-level security attributes. Will override any matching Pod-level attributes. See `kubectl explain pods.spec.containers.securityContext`. |
+| `containerSecurityContext` | [Security Context](k8s1.13.5.md#security-context) | Container-level security attributes. Will override any matching Pod-level attributes. See `kubectl explain pods.spec.containers.securityContext`. Beginning with operator version 4.0.5, the operator will use default content for container-level `securityContext` if no value is specified for this field. More info: https://oracle.github.io/weblogic-kubernetes-operator/security/domain-security/pod-and-container/ |
```
Comment (rosemarymarano) [inline documentation/domains/Cluster.md:59]: -> Beginning with operator version 4.0.5, if no value is specified for this field, the operator will use default content for container-level `securityContext`.
-> Period at the end of the last sentence.

File: documentation/domains/Cluster.md [hunk 3]
```diff
@@ -66,7 +66,7 @@ The specification of the operation of the WebLogic cluster. Required.
...
 | `nodeName` | string | NodeName is a request to schedule this Pod onto a specific Node. If it is non-empty, the scheduler simply schedules this pod onto that node, assuming that it fits the resource requirements. See `kubectl explain pods.spec.nodeName`. |
 | `nodeSelector` | Map | Selector which must match a Node's labels for the Pod to be scheduled on that Node. See `kubectl explain pods.spec.nodeSelector`. |
-| `podSecurityContext` | [Pod Security Context](k8s1.13.5.md#pod-security-context) | Pod-level security attributes. See `kubectl explain pods.spec.securityContext`. |
+| `podSecurityContext` | [Pod Security Context](k8s1.13.5.md#pod-security-context) | Pod-level security attributes. See `kubectl explain pods.spec.securityContext`. Beginning with operator version 4.0.5, the operator will use default content for the pod-level `securityContext` if no value is specified for this field. More info: https://oracle.github.io/weblogic-kubernetes-operator/security/domain-security/pod-and-container/ |
```
Comment (rosemarymarano) [inline documentation/domains/Cluster.md:69]: -> Beginning with operator version 4.0.5, if no value is specified for this field, the operator will use default content for the pod-level `securityContext`.
-> Period at the end of the last sentence.

**Review by rosemarymarano — COMMENTED** [review]
minor edits

**Review by ankedia — APPROVED** [review]
LGTM

File: Dockerfile [hunk 4]
```diff
@@ -43,21 +43,22 @@ RUN set -eux; \
         alternatives --install "/usr/bin/$base" "$base" "$bin" 20000; \
     done; \
     java -Xshare:dump; \
-    useradd -d /operator -M -s /bin/bash -g root -u 1000 oracle; \
+    groupadd -g 1000 oracle; \
+    useradd -d /operator -M -s /bin/bash -g oracle -u 1000 oracle; \
```
Comment (pfmackin) [inline Dockerfile:47]: wondering if there is any possible side effect for domain on PV where existing PV might have existing files with root group permissions that are no longer accessible for user with 1000 GID.

Comment (rjeberhard) [reply Dockerfile:47]: This Dockerfile is for the operator itself and not for WebLogic images. The operator doesn't mount or access PV's for domains.

**Pull request merged**